src/app/components/chat/jh-chat.directive.js

```diff
@@ -47,19 +47,18 @@
         }
 
         // Update the chat messages count
-        for (var i = scope.entriesCount; i < entries.length; i++) {
-          if (entries[i].type === 'chatMsg') {
-            scope.messagesCount++;
-            if (scope.isChatVisible) {
-              scope.lastSeenMessage++;
-            } else {
-              var chatHeader = document.getElementById('chat-header');
-              chatHeader.innerHTML = scope.messagesCount - scope.lastSeenMessage;
-            }
+        entries.slice(scope.entriesCount).filter(function(entry) {
+          // We don't notify the user when the entry is not of type 'chatMsg'
+          return entry.type === "chatMsg";
+        }).forEach(function(entry) {
```

Comment:
About chat header, instead of using `getElementById` and updating it manually, maybe you should use a variable in the scope (`scope.chatHeader`) and use it in the template.

Thinking about it, maybe you could use the same approach when updating CSS classes and all that stuff.

I mean: try to set proper values on the directive's scope and let the template to show the correct values.